perf-event/src/events/msr.rs

```diff
@@ -0,0 +1,68 @@
+use crate::events::Event;
+use c_enum::c_enum;
+use perf_event_open_sys::bindings;
+use std::io;
+use std::sync::atomic::{AtomicU32, Ordering};
+
+// 0 will never be the PMU value for msr.
+// We use it as a flag value to indicate that this has not been initialized.
+static MSR_TYPE: AtomicU32 = AtomicU32::new(0);
+
+
+c_enum! {
+    /// MSR events on all x86 CPUs
+    #[derive(Clone, Copy, Eq, PartialEq, Hash)]
+    pub enum MSRConfig: u64 {
+        /// Time Stamp Counter
+        /// TSC increases at the CPU base frequency.
+        TSC = 0x0,
+        /// Actual Performance Frequency Clock Count
+        APERF = 0x1,
+        /// Maximum Performance Frequency Clock Count
+        /// (APERF / MPERF) * CPU_BASE_FREQUENCY is the running CPU frequency
+        MPERF = 0x2,
```

Comment:
Please structure the doc comments on each item like this:

```
/// First line.
///
/// ... rest of docs
```
The first paragraph shows up in docs search in rustdoc so it should ideally be at most 1-2 lines.